src/DataStructures/Tensor/Expressions/AddSubtract.hpp

```diff
@@ -38,6 +40,81 @@ struct TensorExpression;
+/// \endcode
+/// `input_symm2`:
+/// \code
+/// {2, 2, 1} // S' symmetry, where c and a are symmetric indices
+/// \endcode
+/// returned equivalent `output_symm2`:
+/// \code
+/// {2, 1, 2} // S' symmetry (input_symm2) rearranged to R's index order (abc)
+/// \endcode
+///
+/// One special case scenario to note is when concrete time indices are
+/// involved in the transformation. Consider transforming the symmetry for
+/// some tensor \f$S_{ab}\f$ to the index order of another tensor
+/// \f$R_{btat}\f$. This would be necessary in an expression such as
+/// \f$R_{btat} + S_{ab}\f$. The transformation of the symmetry for \f$S\f$
+/// according to the index order of \f$R\f$ cannot simply be the list of
+/// positions of \f$R\f$'s indices in \f$S\f$' indices, as \f$S\f$ does not
+/// contain all of \f$R\f$'s indices, because it has no time indices. To handle
+/// cases like this, a placeholder value for the position of any time index
+/// must be substituted for an actual position, since one may not exist. In this
+/// example, the proper input transformation (`tensorindex_transformation`)
+/// would need to be
+/// `{1, PLACEHOLDER_VALUE, 0, PLACEHOLDER_VALUE}`, where `PLACEHOLDER_VALUE` is
+/// defined by
+/// `TensorIndexTransformation_detail::time_index_position_placeholder`. `1` and
+/// `0` are the positions of \f$b\f$ and \f$a\f$ in \f$S\f$, and the placeholder
+/// is used for the positions of time indices. In computing the output
+/// transformed symmetry, the function will insert a `0` at each position
+/// where this placeholder is found in the transformation. For example, if
+/// `input_symm2` is `{2, 1}`, the returned output multi-index will be
+/// `{1, 0, 2, 0}`. Note that the symmetry returned by this function is not
+/// necessarily in the canonical form defined by ::Symmetry. This example with
+/// time indices is an example, as `0` is not a permitted ::Symmetry value, and
+/// the canonical form would have increasing symmetry values from right to left.
+/// In addition, even though the time indices in the rearranged symmetry of the
+/// second operand will have the same symmetry value (`0`), this bears no impact
+/// on `get_addsub_symm`'s computation of the symmetry of the tensor resulting
+/// from the addition or subtraction.
+///
+/// \tparam NumIndices2 the number of indices in the second operand
+/// \tparam NumIndices1 the number of indices in the first operand
+/// \param input_symm2 the second operand's symmetry to transform
+/// \param tensorindex_transformation the positions of the first operand's
+/// generic indices in the first operand's generic indices (see example in
+/// details)
+/// \return the the rearranged symmetry of the second operand accoring to the
+/// generic index order of the first operand
+template <size_t NumIndices2, size_t NumIndices1>
+SPECTRE_ALWAYS_INLINE constexpr std::array<std::int32_t, NumIndices1>
+transform_addsub_symm2(const std::array<std::int32_t, NumIndices2>& input_symm2,
```

Comment:
Why the `symm2` instead of just `symm`? If there's a naming collision, I'd suggest making the names clearer :)

Reply:
This is to make it clear that the symmetry being transformed is the symmetry of the 2nd operand. I guess I don't see a reason why it can't be generically defined. I'll update this!

Reply:
ah fair. If you wanted to emphasize that I think doing `..._symm_op2` would be clearer. Up to you. I'm happy with the fixup too :)